docs/source/tasks/language_modeling.mdx

```diff
@@ -0,0 +1,408 @@
+<!--Copyright 2022 The HuggingFace Team. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+the License. You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+specific language governing permissions and limitations under the License.
+-->
+
+# Language modeling
+
+Language modeling predicts words in a sentence. There are two forms of language modeling:
+
+- Causal language modeling: predict the next token in a sequence of tokens, and the model can only attend to tokens on the left.
+- Masked language modeling: predict a masked token in a sequence, and the model can attend to tokens bidirectionally.
+
+This guide will show you how to fine-tune [DistilGPT2](https://huggingface.co/distilgpt2) or [DistilRoBERTa](https://huggingface.co/distilroberta-base) on the [r/askscience](https://www.reddit.com/r/askscience/) subset of the [ELI5](https://huggingface.co/datasets/eli5) dataset for causal and masked language modeling respectively.
```

Comment:
Maybe mention another tokenizer/architecture can be used the same way somewhere? This feels like it only works for DistilGPT2/DistilRoBERTa